### PR TITLE
feat(dynamic-exporter): Add dynamic flow exporter support for hubble

### PR DESCRIPTION
**Pull request by anubhabMajumdar**
# Description

Adding support for dynamic flow exporter in Hubble.
Read more about dynamic flow exporter [here](https://github.com/cilium/cilium/blob/42632319ec752ba246881ed3dd14bc1d0516e4de/Documentation/observability/hubble/configuration/export.rst).

## Checklist

- [x] I have read the [contributing documentation](https://retina.sh/docs/contributing).
- [x] I signed and signed-off the commits (`git commit -S -s ...`). See [this documentation](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification) on signing commits.
- [x] I have correctly attributed the author(s) of the code.
- [x] I have tested the changes locally.
- [x] I have followed the project's style guidelines.
- [x] I have updated the documentation, if necessary.
- [x] I have added tests, if applicable.

## Screenshots (if applicable) or Testing Completed

```bash
$ kno aks-nodepool1-10725549-vmss000000
root@aks-nodepool1-10725549-vmss000000:/# tail -f /var/run/retina/hubble/events.log
...
{"flow":{"time":"2024-11-19T22:35:33.343220551Z","verdict":"FORWARDED","IP":{"source":"192.168.2.38","destination":"10.224.0.4","ipVersion":"IPv4"},"l4":{"TCP":{"source_port":46518,"destination_port":10250,"flags":{"ACK":true}}},"source":{"ID":31432,"identity":31432,"namespace":"kube-system","pod_name":"konnectivity-agent-7d494c7d79-7pm4z"},"destination":{"ID":1,"identity":1,"labels":["reserved:host"]},"Type":"L3_L4","event_type":{"type":4,"sub_type":3},"traffic_direction":"EGRESS","trace_observation_point":"TO_STACK","is_reply":false,"Summary":"TCP Flags: ACK:true","extensions":{"@type":"type.googleapis.com/utils.RetinaMetadata","bytes":66}},"time":"2024-11-19T22:35:33.343220551Z"}
{"flow":{"time":"2024-11-19T22:35:33.343337550Z","verdict":"FORWARDED","IP":{"source":"192.168.2.38","destination":"10.224.0.4","ipVersion":"IPv4"},"l4":{"TCP":{"source_port":46518,"destination_port":10250,"flags":{"ACK":true}}},"source":{"ID":31432,"identity":31432,"namespace":"kube-system","pod_name":"konnectivity-agent-7d494c7d79-7pm4z"},"destination":{"ID":1,"identity":1,"labels":["reserved:host"]},"Type":"L3_L4","event_type":{"type":4,"sub_type":3},"traffic_direction":"EGRESS","trace_observation_point":"TO_STACK","is_reply":false,"Summary":"TCP Flags: ACK:true","extensions":{"@type":"type.googleapis.com/utils.RetinaMetadata","bytes":66}},"time":"2024-11-19T22:35:33.343337550Z"}
root@aks-nodepool1-10725549-vmss000000:/# ls -la /var/run/retina/hubble
total 51480
drwxr-xr-x 2 root root      160 Nov 19 22:35 .
drwxr-xr-x 3 root root       60 Nov 19 01:53 ..
-rw-r--r-- 1 root root 10485267 Nov 19 16:32 events-2024-11-19T16-32-15.656.log
-rw-r--r-- 1 root root 10485050 Nov 19 16:49 events-2024-11-19T16-49-23.125.log
-rw-r--r-- 1 root root 10485079 Nov 19 17:06 events-2024-11-19T17-06-33.124.log
-rw-r--r-- 1 root root 10485306 Nov 19 17:23 events-2024-11-19T17-23-47.045.log
-rw-r--r-- 1 root root 10485271 Nov 19 22:35 events-2024-11-19T22-35-48.976.log
-rw-r--r-- 1 root root   284620 Nov 19 22:36 events.log
root@aks-nodepool1-10725549-vmss000000:/#
```

---

Please refer to the [CONTRIBUTING.md](../CONTRIBUTING.md) file for more information on how to contribute to this project.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -577,7 +583,7 @@ quick-deploy:
 .PHONY: quick-deploy-hubble
 quick-deploy-hubble:
 	$(MAKE) helm-uninstall || true
-	$(MAKE) helm-install-without-tls HELM_IMAGE_TAG=$(TAG)-linux-amd64
+	$(MAKE) helm-install-without-tls HELM_IMAGE_TAG=$(TAG)-linux-amd64 ENABLE_DYNAMIC_EXPORT=true
```
Comment (nddq) [inline Makefile:586]: do we want to set ENABLE_DYNAMIC_EXPORT to true by default here?

File: controller/Dockerfile [hunk 2]
```diff
@@ -116,8 +116,8 @@ ENTRYPOINT ["./retina/initretina"]
 
 
 # agent final image
-# mcr.microsoft.com/cbl-mariner/distroless/minimal:2.0
-# mcr.microsoft.com/cbl-mariner/distroless/minimal@sha256:63a0a70ceaa1320bc6eb98b81106667d43e46b674731ea8d28e4de1b87e0747f
+# For debug: mcr.microsoft.com/cbl-mariner/distroless/debug:2.0
```
Comment (nddq) [inline controller/Dockerfile:119]: change this comment back to the original one? since it was identifying which image is being used for the actual agent

**Review by nddq — COMMENTED** [review]
some nit comments but lgtm otherwise

File: Makefile (hunk 1 above)
Comment (anubhabMajumdar) [reply Makefile:586]: `quick-deploy*` is for dev purposes, so IMO it should enable all features.

File: controller/Dockerfile (hunk 2 above)
Comment (anubhabMajumdar) [reply controller/Dockerfile:119]: Don't want to keep updating this comment. I want to add the debug image as it's very useful for debugging purpose during development.

**Pull request closed without merging**